orx/grantlee/0.2/OrxJs.qs

```diff
@@ -0,0 +1,272 @@
+var buffer = '';
+
+function append() {
+    for (var i = 0; i < arguments.length; i++) {
+        if (arguments[i] !== undefined) {
+            buffer += arguments[i] + '\n';
+        }
+    }
+    if (arguments.length == 0) {
+        buffer += '\n';
+    }
+}
+
+function friendlyName(name) {
+    name = name.replace(/[^a-z0-9]/gi, ' ');        // Keep alpha-numeric
+    name = name[0].toUpperCase() + name.slice(1); // Proper case
+    return name;
+}
+
+function getId(name) {
+    // 'big enemy/hard-boss/boss_01' -> ['big', 'enemy', 'hard', 'boss', 'boss', '01']
+    var parts = name.split(/\/|_|-| /g);
+    
+    for (var i = 0; i < parts.length; i++) {
+        parts[i] = friendlyName(parts[i]);
+    }
+    
+    // Optimize any sequences of the same identifier.
+    // E.g. ['Boss', 'Boss', 'Boss'] -> 'Boss'
+    var clean = parts[0];
+    for (var i = 1; i < parts.length; i++) {
+        if (parts[i] !== parts[i - 1]) {
+            clean += parts[i];
+        }
+    }
+    
+    // Now we have an identifier suitable for .ini files.
+    // E.g. 'BigEnemyHardBoss01'
+    
+    return clean;
+}
+
+function section(id) {
+    return '[' + id + ']';
+}
+
+function frameSection(root, sprite, id, parentId) {
+    var frameId = id;
+    if (parentId !== undefined) {
+        frameId += '@' + parentId;
```

Comment:
I might be wrong as I didn't run the exporter, but it seems this will conflict with the rule for frame sections (which actually also applies to animation sections).
From `CreationTemplate.ini`:
```ini
[AnimationTemplate1]
; NB: Do not use manual inheritance at the section level as it will be overridden at runtime: its parent will then become the animation section;
```

Reply:
That line is there to handle exactly that. Regular textures inherit from the section with the png. E.g.:

```ini
[Sprites]
Texture       = sprites.png
TextureSize   = (2048, 2048)

[MyAnimationSet]
Texture       = sprites.png
; ...

[MyBoss@Sprites]
TextureOrigin = (13, 37)
TextureSize   = (42, 42)
Pivot         = center
```